### PR TITLE
Do not allow local urls for the report bug link

### PR DESCRIPTION
**Pull request by danidoni**
This is how it looks like when the url is local
Screenshot from 2024-09-20 11-20-42

And this is how it looks like when the url is not local
Screenshot from 2024-09-20 11-53-31

### CODE REVIEW

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -1368,6 +1369,12 @@ def convert_to_symsync
     BackendPackage.where(package_id: id).delete_all
     decline_requests_with_self_as_target("The package '#{project.name} / #{name}' is now maintained at #{scmsync}")
   end
+
+  def report_bug_url_is_not_local
+    return true unless report_bug_url
```
Comment (eduardoj) [inline src/api/app/models/package.rb:1374]: Isn't this one an existence validation?

Comment (eduardoj) [reply src/api/app/models/package.rb:1374]: It is not.

Comment (danidoni) [reply src/api/app/models/package.rb:1374]: Well no, this is a guard clause, if not, the line below crashes

**Review by eduardoj — APPROVED** [review]

**Review by saraycp — APPROVED** [review]

File: src/api/app/models/package.rb [hunk 2]
```diff
@@ -104,6 +104,7 @@ class Package < ApplicationRecord
   validates :url, length: { maximum: 255 }
   validates :description, length: { maximum: 65_535 }
   validates :report_bug_url, length: { maximum: 8192 }
+  validate :report_bug_url_is_not_local
```
Comment (saraycp) [inline src/api/app/models/package.rb:107]: What about `report_bug_url_is_external`?

Comment (danidoni) [reply src/api/app/models/package.rb:107]: Could work as well, yes

Comment (danidoni) [reply src/api/app/models/package.rb]: Done

**Review by eduardoj — APPROVED** [review]

File: src/api/app/models/package.rb [hunk 3]
```diff
@@ -1368,6 +1369,12 @@ def convert_to_symsync
     BackendPackage.where(package_id: id).delete_all
     decline_requests_with_self_as_target("The package '#{project.name} / #{name}' is now maintained at #{scmsync}")
   end
+
+  def report_bug_url_is_external
+    return true unless report_bug_url
+
+    errors.add(:report_bug_url, 'Local urls are not allowed') if report_bug_url.include?(Configuration.obs_url)
```
Comment (hennevogel) [inline src/api/app/models/package.rb:1376]: Why about urls without protocol or host?

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/app/models/package.rb (hunk 3 above)
Comment (danidoni) [reply src/api/app/models/package.rb:1376]: Oh right, relative urls...

Comment (danidoni) [reply src/api/app/models/package.rb]: Done

File: src/api/spec/models/package_spec.rb [hunk 4]
```diff
@@ -781,4 +781,34 @@
...
+
+  describe '#report_bug_url' do
+    # Locally the configuration returns https://unconfigured.openbuildservice.org
+    # as host so we use a better host
+    before { allow(Configuration).to receive(:obs_url).and_return('https://localhost:3000') }
+
+    context 'url is external' do
+      let(:package) { build(:package, report_bug_url: 'https://example.com') }
+
+      it { expect(package).to be_valid }
```
Comment (hennevogel) [inline src/api/spec/models/package_spec.rb:793]: How do you know why it is or isn't valid? Could be any validation...

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/spec/models/package_spec.rb (hunk 4 above)
Comment (danidoni) [reply src/api/spec/models/package_spec.rb:793]: The package is build valid, and I only change the `report_bug_url` so if it's invalid it has to be this change

Comment (danidoni) [reply src/api/spec/models/package_spec.rb]: Fixed

**Review by hennevogel — APPROVED** [review]

**Pull request merged**